lib/src/core/util/bloc_singleton_dependencies.dart

```diff
@@ -0,0 +1,69 @@
+import '../../data/datasource/local/movie_database.dart';
+import '../../data/datasource/remote/api_genres_service.dart';
+import '../../data/datasource/remote/api_movie_service.dart';
+import '../../data/repository/genre_repository_impl.dart';
+import '../../data/repository/movie_repository_impl.dart';
+import '../../domain/usecase/implementation/genres_use_case.dart';
+import '../../domain/usecase/implementation/movies_use_case.dart';
+import '../../presentation/bloc/genres_bloc.dart';
+import '../../presentation/bloc/movies_bloc.dart';
+import '../../presentation/bloc/trailers_screen_bloc.dart';
+import 'api_constants.dart';
+
+class BlocSingletonDependencies {
+  static final databaseInstance =
+      $FloorAppDatabase.databaseBuilder('app_database.db').build();
+
+  static MoviesBloc moviesBloc = MoviesBloc(
+    popularMoviesUseCase: MoviesUseCase(
+      movieRepository: MovieRepositoryImpl(
+        movieService: ApiMovieService(
```

Comment:
this classes should receive the endpoint in the fetch method instead of in the constructor